Add the ability for Genesis to automatically extract AppRole from Exodus

Proplex opened the pull request:
This goes in hand with https://github.com/genesis-community/concourse-genesis-kit/pull/38 so that operators no longer have to extract AppRole credentials for pipeline setup.

jhunt commented on lib/Genesis/CI/Legacy.pm:
```diff
@@ -610,7 +610,9 @@ pipeline:
-
+  vault:
+    role:   (( vault "secret/exodus/ci/genesis_pipelines:approle_id" ))
+    secret: (( vault "secret/exodus/ci/genesis_pipelines:approle_secret" ))
```
The outstanding item on the other related PR is to rename `genesis_pipelines` to `genesis-pipelines`; we'll need to remember to propagate the update here as well.

jhunt commented on lib/Genesis/CI/Legacy.pm:
```diff
@@ -1119,8 +1121,8 @@ EOF
-            VAULT_ROLE_ID:        $pipeline->{pipeline}{vault}{role}
-            VAULT_SECRET_ID:      $pipeline->{pipeline}{vault}{secret}
+            VAULT_ROLE_ID:        (( grab pipeline.vault.role ))
+            VAULT_SECRET_ID:      (( grab pipeline.vault.secret ))
```
Since we're no longer using these, we should remove all the other references to handling them (in other parts of Legacy.pm), and warn if they are present.

Proplex replied:
I've kept them here in case an operator wanted to override what we do. i.e. a security team doesn't like our policies or whatnot, and instead they'd prefer to manage their own AppRole/policies. By default, if `pipeline.vault.*` in `ci.yml` takes precedent if they're defined. Otherwise, fallback to our in-Vault, Concourse-kit-generated credentials.

jhunt replied:
love it!

Proplex commented:
Adjusted for feedback. I've removed references to `$P->{pipeline}{vault}{role}` and `$P->{pipeline}{vault}{secret}`.  I've also updated documentation that refers to the new addon command in the Concourse pipeline.
